### ISSUE TITLE
Rule Change: indent

### ISSUE THREAD

**Issue by alexanderankin** [eslint/eslint#16569]
### What rule do you want to change?

indent

### What change to do you want to make?

Implement suggestions

### How do you think the change should be implemented?

Other

### Example code

```js
@Component({
    selector: 'process-status'
})
export class Abc implements OnInit {
  ngOnInit(): void {}
}
```

### What does the rule currently do for this code?

The rule will cause people to google and discover that eslint maintainers (maybe ts extension eslint maintainers?) use prettier instead of eslint for this need, and that is why it is broken and non-functional. I would like it to instead raise the level of experience of all developers by directing them to use the right tools for the job (and critically, fostering goodwill towards eslint in the community, instead of illwill).

it instead renders the below:

```typescript
@Component({
    selector: 'process-status'
    }) /// ?????
export class Abc implements OnInit {
  ngOnInit(): void {}
}
```

### What will the rule do after it's changed?

print a warning if it is explicitly disabled that it is deprecated for removal and will not do anything (not fix with `--fix` or cause non-0 exit codes with "error" level setting). Or, that this is just disabled by default?

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

thanks for your time and consideration!!

**Comment by mdjermanovic** [comment]
Hi @alexanderankin, thanks for the issue!

Are you referring to the core `indent` rule or  `@typescript-eslint/indent` rule? The core `indent` rule is not deprecated, and it's functional for syntax defined in the latest version of [ECMAScript Language Specification](https://tc39.es/ecma262/).

**Comment by alexanderankin** [comment]
Hi @mdjermanovic thanks for the reply - do you know who i might contact about the rule described here? https://typescript-eslint.io/rules/indent/

**Comment by alexanderankin** [comment]
oh, i guess it links to a repo other than this one, ok thats my bad, sorry
